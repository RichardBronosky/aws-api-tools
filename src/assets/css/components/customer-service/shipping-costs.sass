@import '../../mixins'
@import '../../variables'

.shipping-costs
  background-color: white
  padding: 1rem 3rem
  margin: 0 auto
  max-width: 1200px
  .category
    padding: 1rem 0
    .category-header
      text-transform: uppercase
      color: $dark-grey
      font-size: $txt-medium-up
      padding: 0
    .category-text
      font-size: $txt-medium
      color: #666
      margin: 1rem 0
    .error-message
      color: $red
      font-size: 1rem
      margin: 1rem 0
    .input-wrapper
      display: flex      
      .zip-input
        padding: 0.5rem
        margin-right: 10px
        font-size: 1rem
      .submit-btn
        background-color: $primary
        color: white
        border: 0
        text-align: center
        font-size: 1rem
        text-transform: uppercase
        padding: 0.75rem