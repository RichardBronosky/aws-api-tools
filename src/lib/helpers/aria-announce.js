const formatMsg = msg => {
  return '<div aria-live="assertive" aria-atomic="true">' + msg + '</div>'
}

export const taskDone = (fn, time, id) => {
  let timers = {}
  if (typeof fn == 'function') {
    if (timers[id]) clearTimeout(timers[id])
    timers[id] = setTimeout(fn, time)
  }
}

export const announce = (msg, form, ele, delay) => {
  if (typeof ele == 'object') {
    msg += ele.innerText.replace(/\n/g, ' ')
  }
  if (document) {
    document.getElementById('aria-announce').innerHTML = ''
    taskDone(
      () => {
        document.getElementById('aria-announce').innerHTML = formatMsg(msg)
      },
      delay || 500,
      'announceMessage'
    )
    if (form) {
      taskDone(
        () => {
          form.querySelectorAll('[aria-invalid="true"]')[0].focus()
        },
        1000,
        'focusElement'
      )
    }
  }
}
