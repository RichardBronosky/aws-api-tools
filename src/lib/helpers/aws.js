const AWS = require('aws-sdk')
const { loopWhile } = require('deasync')
const client = new AWS.SecretsManager({ region: 'us-east-1' })
const { GET_SECRETS } = process.env

const getSecretNames = async () => {
  const resp = await client.listSecrets({ MaxResults: 100 }).promise()
  const secrets = resp.SecretList.map(s => s.Name)
  return secrets
}

const tryParse = stringValue => {
  try {
    return JSON.parse(stringValue)
  } catch (err) {
    return {}
  }
}

const getSecret = async name => {
  const resp = await client.getSecretValue({ SecretId: name }).promise()
  return tryParse(resp.SecretString)
}

exports.getSecrets = () => {
  let secrets = {}
  let done = false
  if (GET_SECRETS) {
    getSecretNames()
      .then(names => Promise.all(names.map(n => getSecret(n))))
      .then(secretObjects => {
        secretObjects.forEach(s => Object.assign(secrets, s))
        done = true
      })
      .catch(err => {
        done = true
        console.log('\nERROR (/helpers/aws.js): ' + err)
      })
    loopWhile(() => !done)
  }
  return secrets
}
