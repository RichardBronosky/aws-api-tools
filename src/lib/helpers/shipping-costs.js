import { navigate } from 'gatsby'
import { getLocation } from '..//services/location'
import { validateZip } from './string-helper'

export const handleSubmit = (e, zipCode, errorMessage, setState) => {
  e.preventDefault()
  if (validateZip(zipCode)) {
    if (errorMessage) {
      setState({ errorMessage: null })
    }
    getLocation(zipCode)
      .then(data => {
        const deliveryZone =
          data.success && data.response && data.response.delivery_zone ? data.response.delivery_zone : null
        if (deliveryZone) {
          if (deliveryZone.toLowerCase() === 'ups') {
            setState({
              errorMessage:
                'Sorry, we do not deliver furniture to your area. Select accessories are available and will ship via UPS, as indicated.',
            })
          } else {
            navigate(`/delivery-charges-zone-${ deliveryZone.toLowerCase() }`)
          }
        } else {
          setState({ errorMessage: 'Sorry, something went wrong. Please try again with a valid zip code.' })
        }
      })
      .catch(() => setState({ errorMessage: 'Sorry, something went wrong. Please try again with a valid zip code.' }))
  } else {
    setState({ errorMessage: 'Please enter a valid zip code.' })
  }
}
