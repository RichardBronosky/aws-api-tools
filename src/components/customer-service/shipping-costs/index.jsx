import React from 'react'
import { getCurrentLocation } from '../../../lib/helpers/geo-location'
import ShippingCosts from './shipping-costs'

export default class ShippingCostsWrapper extends React.Component {
  state = {
    zipCode: getCurrentLocation().zip,
    errorMessage: null,
  }

  setShippingCostsState = state => this.setState(state)

  render() {
    const { zipCode, errorMessage } = this.state
    return <ShippingCosts zipCode={ zipCode } errorMessage={ errorMessage } setState={ this.setShippingCostsState } />
  }
}
