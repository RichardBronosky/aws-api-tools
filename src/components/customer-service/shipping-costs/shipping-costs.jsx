import React from 'react'
import { handleSubmit } from '../../../lib/helpers/shipping-costs'
import '../../../assets/css/components/customer-service/shipping-costs.sass'

export default ({ zipCode, errorMessage, setState }) => (
  <div className="shipping-costs">
    <div className="category">
      <h1 className="category-header">ESTIMATED SHIPPING COSTS</h1>
      <p className="category-text">
        Enter the zip code where the furniture is being DELIVERED to see the delivery costs to that area
      </p>
      { errorMessage && <div className="error-message">{ errorMessage }</div> }
      <form className="input-wrapper" onSubmit={ e => handleSubmit(e, zipCode, errorMessage, setState) }>
        <input
          className="zip-input"
          placeholder="Enter ZIP code"
          value={ zipCode }
          onChange={ e => setState({ zipCode: e.target.value }) }
        />
        <button className="submit-btn" type="submit">
          Go
        </button>
      </form>
    </div>
    <div className="category">
      <img
        src="//images.ctfassets.net/n86sd31cg9pg/2d4uJ6HKg8gIkSGoEYqIyw/aeadf3096828b82337e688714eb4b480/delivery-map2-bg.png"
        role="presentation"
        alt="Delivery Trucks"
      />
    </div>
    <div className="category">
      <h1 className="category-header">WHERE WE DELIVER</h1>
      <p className="category-text">
        Rooms To Go will deliver and set up your furniture purchase to the following states: Alabama, Florida, Georgia,
        South Carolina, North Carolina, Tennessee, Mississippi, Texas, Virginia and Kentucky. In addition, we deliver to
        many areas in Oklahoma, Massachusetts, New York, New Jersey, Pennsylvania, Maryland, Delaware, Connecticut,
        Louisiana, Rhode Island and Washington, D.C.
      </p>
    </div>
    <div className="category">
      <h1 className="category-header">FREE UPS DELIVERY</h1>
      <p className="category-text">
        While furniture delivery is not available to every state, many home accessories on roomstogo.com are shipped via
        UPS Standard Ground shipping. This merchandise is clearly identified and will be shipped anywhere in the
        Contiguous U.S. often at no shipping cost to the consumer.
      </p>
    </div>
    <div className="category">
      <h1 className="category-header">QUESTIONS AFTER DELIVERY</h1>
      <p className="category-text">Call Customer Care Center at 1-800-766-6786 with any concerns.</p>
      <p className="category-text">Hours:</p>
      <p className="category-text">Monday - Friday 8:00am - 8:00pm Eastern</p>
      <p className="category-text">Saturday - 8:00am - 6:00pm Eastern</p>
    </div>
  </div>
)
