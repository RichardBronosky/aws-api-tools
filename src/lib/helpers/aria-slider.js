import { taskDone } from './aria-announce'

export const sliderfix = (slider, type) => {
  let ele = slider

  const add = arr => {
    for (let ii = 0; ii < arr.length; ii++) {
      let anchor = arr[ii]
      anchor.setAttribute('tabindex', '-1')
      anchor.setAttribute('aria-hidden', 'true')
    }
  }

  const remove = arr => {
    for (let ii = 0; ii < arr.length; ii++) {
      let anchor = arr[ii]
      anchor.removeAttribute('tabindex')
      anchor.removeAttribute('aria-hidden')
    }
  }

  const check = () => {
    if (ele) {
      let parent = ele.current
      if (parent) {
        let slides = parent.querySelectorAll('.slick-slide')
        for (let i = 0; i < slides.length; i++) {
          let slide = slides[i]

          if (slide.classList.contains('slick-active')) {
            remove(slide.querySelectorAll('a'))
            remove(slide.querySelectorAll('button'))
          } else {
            add(slide.querySelectorAll('a'))
            add(slide.querySelectorAll('button'))
          }
        }
      }
    }
  }

  if (ele && ele.current) {
    taskDone(
      () => {
        check()
      },
      type == 'init' ? 1 : 200,
      'adjustSlides'
    )
  }
}
