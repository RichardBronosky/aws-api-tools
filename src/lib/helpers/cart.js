import { fetchProductBySku, fetchProductWarehouseAvailability } from '../services/product'
import { productPrice, productOnSale, availabilityStockMessage, fetchProductOrGC } from './product'
import { store } from '../../redux/store'
import { setOrder, setCheckoutStep } from '../../redux/modules/checkout'
import {
  addProductToCart,
  updateCartProductQuantity,
  removeProductFromCart,
  setCart,
  addActiveAddon,
  removeActiveAddon as removeActiveAddonRedux,
} from '../../redux/modules/cart'
import { analyticsProduct, addToDataLayer } from './google-tag-manager'
import { saveLocalStorage, getFromBrowserStorage } from './storage'
import { taskDone, announce } from './aria-announce'
import { clearCheckoutState, getOrder, getLineItems } from './checkout/global'
import { getRegionZone, getCurrentLocation } from './geo-location'
import { createOrder, getOrder as fetchOrder, updateLineItems } from '../services/checkout'

export const getCart = () => {
  return store.getState().cart.cart
}

export const fetchAndAddActiveAddon = (product, cartIndex, addon) => {
  fetchProductBySku(addon.sku).then(fetchedAddon =>
    store.dispatch(addActiveAddon(product.sku, { ...fetchedAddon, quantity: addon.quantity }, cartIndex))
  )
}

export const removeActiveAddon = (product, addonSku, cartIndex) => {
  store.dispatch(removeActiveAddonRedux(product.sku, addonSku, cartIndex))
}

export const getLineItemQuantity = () => {
  let count = 0
  let count2 = 0
  const cart = getCart()
  if (cart && cart.cartItems && cart.cartItems.length > 0) {
    for (let i = 0, n = cart.cartItems.length; i < n; i++) {
      count += parseInt(cart.cartItems[i].quantity)
      if (cart.cartItems[i].activeAddons) {
        for (let x = 0, y = cart.cartItems[i].activeAddons.length; x < y; x++) {
          count += parseInt(cart.cartItems[i].quantity * cart.cartItems[i].activeAddons[x].quantity)
        }
      }
      if (cart.cartItems[i].product && cart.cartItems[i].product.addon_items) {
        const requiredAddons = cart.cartItems[i].product.addon_items.filter(addon => addon.addon_required)
        if (requiredAddons.length > 0) {
          for (let x = 0, y = requiredAddons.length; x < y; x++) {
            if (
              !cart.cartItems[i].activeAddons ||
              (cart.cartItems[i].activeAddons &&
                cart.cartItems[i].activeAddons.filter(addon => addon.sku === requiredAddons[x].sku).length < 1)
            ) {
              count2 += cart.cartItems[i].quantity * requiredAddons[x].quantity
            }
          }
        }
      }
    }
  }
  return { cart: count, total: count + count2 }
}

export const getCartItemQuantity = sku => {
  const cart = getCart()
  if (cart && cart.cartItems && cart.cartItems.length > 0) {
    for (let i = 0, n = cart.cartItems.length; i < n; i++) {
      if (sku === cart.cartItems[i].sku) {
        return cart.cartItems[i].quantity
      }
    }
  }
  return 0
}

export function addToCart(product, price, activeAddons) {
  clearCheckoutState()
  let cart = getCart()
  let cartItems = cart && cart.cartItems ? cart.cartItems : []
  let alreadyInCart = false
  let existingProduct
  for (let i = 0; i < cartItems.length; i++) {
    if (
      cartItems[i].sku === product.sku &&
      (JSON.stringify(activeAddons) === JSON.stringify(cartItems[i].activeAddons) ||
        (!activeAddons && !cartItems[i].activeAddons))
    ) {
      alreadyInCart = true
      existingProduct = cartItems[i]
    }
  }
  if (!alreadyInCart) {
    store.dispatch(
      addProductToCart({
        sku: product.sku,
        quantity: 1,
        product: cartProduct(product),
        price: price,
        activeAddons: activeAddons,
      })
    )
  } else {
    store.dispatch(updateCartProductQuantity(existingProduct.sku, parseInt(existingProduct.quantity) + 1))
  }
}

export function removeFromCart(product, index) {
  clearCheckoutState()
  let cart = getCart()
  if (cart && cart.cartItems) {
    let cartItems = cart.cartItems
    for (let i = 0; i < cartItems.length; i++) {
      if (product.sku === cartItems[i].sku && index === i) {
        const quantity = cartItems[i].quantity
        fetchProductOrGC(product.sku).then(data => {
          window.dataLayer.push({
            event: 'ee_remove',
            ecommerce: { remove: { products: [analyticsProduct(data, quantity)] } },
          })
        })
        store.dispatch(removeProductFromCart(product.sku))
      }
    }
    let cartTotal = getCartTotal(cart)
    announce('Item removed. New cart total: $' + cartTotal)
  }
  if (cart && cart.cartItems.length == 0) {
    try {
      taskDone(
        () => {
          const ele = document.getElementById('empty-cart-continue')
          if (ele) {
            ele.focus()
          }
        },
        500,
        'focusEmpty'
      )
    } catch (e) {}
  }
}

export function getCartTotal(cart) {
  let subtotal = 0
  if (cart && cart.cartItems) {
    if (cart && cart.cartItems) {
      let cartItems = cart.cartItems
      for (let i = 0; i < cartItems.length; i++) {
        subtotal = subtotal + cartItems[i].quantity * cartItems[i].price
        if (cartItems[i].activeAddons) {
          for (let x = 0, n = cartItems[i].activeAddons.length; x < n; x++) {
            subtotal =
              subtotal +
              productPrice(cartItems[i].activeAddons[x]) * cartItems[i].activeAddons[x].quantity * cartItems[i].quantity
          }
        }
      }
    }
  }
  return subtotal
}

export const verifyAndUpdateCart = async cartIn => {
  const cart = { ...cartIn }
  if (cart && cart.cartItems && cart.cartItems.length > 0) {
    const cartItems = cart.cartItems
    const region = getRegionZone().region
    const price_zone = getRegionZone().zone
    const priceString = region + '_' + price_zone
    for (let i = 0, n = cartItems.length; i < n; i++) {
      if (cartItems[i]) {
        const product = await fetchProductOrGC(cartItems[i].sku)
        cartItems[i].product = cartProduct(product)
        const regionSalePrice = product.pricing[priceString + '_sale_price']
        const regionListPrice = product.pricing[priceString + '_list_price']
        if (regionSalePrice) {
          cartItems[i].price = regionSalePrice
        } else if (regionListPrice) {
          cartItems[i].price = regionListPrice
        } else {
          cartItems[i].price = product.pricing.default_price
        }
        if (i === n - 1) {
          cart.cartItems = cartItems
        }
      }
    }
  }
  saveLocalStorage('cart', cart)
}

export const checkProductRegionAvailability = async regionIn => {
  const rtg_location = getCurrentLocation()
  const region = regionIn ? regionIn : rtg_location ? rtg_location.region : 'FL'
  let cart = getCart()
  if (!cart || (cart && !cart.cartItems) || (cart && cart.cartItems && cart.cartItems.length < 1)) {
    cart = getFromBrowserStorage('local', 'cart')
  }
  if (cart && cart.cartItems && region) {
    let productPromises = []
    for (let i = 0; i < cart.cartItems.length; i++) {
      productPromises.push(
        fetchProductOrGC(cart.cartItems[i].sku).then(data => {
          let newData = data
          const product = cart.cartItems[i] && cart.cartItems[i].product
          if (
            rtg_location &&
            product &&
            (rtg_location.region !== 'OOM' ||
              ((product.delivery_type === 'O' || product.delivery_type === 'U') && rtg_location.region === 'OOM')) &&
            product.category !== 'gift-card'
          ) {
            return fetchProductWarehouseAvailability(data.sku, rtg_location.distribution_index, rtg_location.state)
              .then(data => {
                const stockMessage = availabilityStockMessage(data, product, rtg_location)
                if (
                  stockMessage === 'Discontinued' ||
                  stockMessage === 'Available Soon' ||
                  stockMessage === 'Out of Stock'
                ) {
                  newData.stock = false
                } else {
                  newData.stock = true
                }
                return newData
              })
              .catch(() => {
                newData.stock = false
                return newData
              })
          } else {
            newData.stock = true
            return newData
          }
        })
      )
    }
    return Promise.all(productPromises).then(products => {
      let skusNotAvailable = []
      for (let i = 0, n = products.length; i < n; i++) {
        if (products[i].category !== 'gift-card') {
          if (!products[i].stock) {
            skusNotAvailable.push({
              sku: products[i].sku,
              reason: 'stock',
            })
          } else if (
            !products[i].catalog_availability[region] &&
            products[i].delivery_type !== 'O' &&
            products[i].delivery_type !== 'U'
          ) {
            skusNotAvailable.push({
              sku: products[i].sku,
              reason: 'region',
            })
          }
        }
      }
      return skusNotAvailable
    })
  } else {
    return []
  }
}

export const getProductsFromCart = cart => {
  const cartObj = cart || getCart()
  let products = []
  let productPromises = []
  if (cartObj && cartObj.cartItems) {
    if (cartObj.cartItems.length > 0) {
      for (let i = 0; i < cartObj.cartItems.length; i++) {
        productPromises.push(fetchProductOrGC(cartObj.cartItems[i].sku))
      }
    } else {
      for (let item in cartObj.cartItems) {
        cartObj.cartItems[item]['product']['quantity'] = cartObj.cartItems[item]['quantity']
        products.push(cartObj.cartItems[item]['product'])
      }
    }
  }
  return Promise.all(productPromises)
}

export const getRoomsToGoDeliveryItems = cartItems => {
  let deliveryItems = []
  for (let i = 0, n = cartItems.length; i < n; i++) {
    let deliveryTypeBlacklist = ['O', 'U', 'T']
    if (!deliveryTypeBlacklist.includes(cartItems[i].product.delivery_type)) {
      deliveryItems.push(cartItems[i])
    }
  }
  return deliveryItems
}

export const getOtherDeliveryItems = cartItems => {
  let deliveryItems = []
  for (let i = 0, n = cartItems.length; i < n; i++) {
    let deliveryTypeWhitelist = ['O', 'U', 'T']
    if (deliveryTypeWhitelist.includes(cartItems[i].product.delivery_type)) {
      deliveryItems.push(cartItems[i])
    }
  }
  return deliveryItems
}

export const cartProduct = (product, quantity) => {
  if (product) {
    let newProduct = product
    const loc = getCurrentLocation()
    if (product.addon_items && product.addon_items.length > 0) {
      for (let i = 0, n = product.addon_items.length; i < n; i++) {
        delete newProduct.addon_items[i]['catalog_availability']
        for (let key in product.addon_items[i].pricing) {
          if (!key.includes(`${ loc.region }_${ loc.price_zone }`) && !key.includes('default')) {
            delete newProduct.addon_items[i].pricing[key]
          }
        }
      }
    }
    if (product.items_in_room) {
      if (product.items_in_room.FL || product.items_in_room.SE || product.items_in_room.TX) {
        for (let key in product.items_in_room) {
          if (key.includes(loc.region)) {
            newProduct.items_in_room = newProduct.items_in_room[key]
          }
        }
      }
    }
    return {
      title: newProduct.title,
      sku: newProduct.sku,
      price: productPrice(newProduct),
      list_price: productPrice(newProduct, true),
      unitPrice: newProduct.unitPrice,
      delivery_type: newProduct.delivery_type,
      category: newProduct.category,
      quantity: newProduct.quantity || quantity,
      free_shipping: newProduct.free_shipping,
      on_sale: productOnSale(newProduct),
      primary_image: newProduct.primary_image,
      addon_items: newProduct.addon_items,
      package_skus: newProduct.package_skus,
      items_in_room: newProduct.items_in_room,
      shipping_cost_code: newProduct.shipping_cost_code,
      strikethrough: newProduct.strikethrough,
    }
  } else {
    return null
  }
}

export const compareLineItems = (cartLineItems, orderLineItems) => {
  for (let i = 0, n = cartLineItems.length; i < n; i++) {
    const orderItems = orderLineItems.filter(orderItem => orderItem.sku === cartLineItems[i].sku)
    let quantity = 0
    for (let x = 0, y = orderItems.length; x < y; x++) {
      quantity += orderItems[x].quantity
    }
    if (quantity !== cartLineItems[i].quantity) {
      return false
    }
  }
  return true
}

export const setNewQuantity = (event, cart, sku, activeAddons) => {
  if (cart && cart.cartItems) {
    let cartItems = cart.cartItems
    for (let i = 0; i < cartItems.length; i++) {
      const quantity = parseInt(event.target.value)
      const productQuantity = cartItems[i].quantity
      if (cartItems[i].sku === sku && JSON.stringify(activeAddons) === JSON.stringify(cartItems[i].activeAddons)) {
        fetchProductOrGC(sku).then(data => {
          if (productQuantity < quantity) {
            window.dataLayer.push({
              event: 'ee_add',
              ecommerce: { add: { products: [analyticsProduct(data, quantity - productQuantity)] } },
            })
          } else if (productQuantity > quantity) {
            window.dataLayer.push({
              event: 'ee_remove',
              ecommerce: { remove: { products: [analyticsProduct(data, productQuantity - quantity)] } },
            })
          }
        })
        store.dispatch(updateCartProductQuantity(cartItems[i].sku, event.target.value))
        addToDataLayer('click', 'cart', 'change quantity', `${ sku }, ${ event.target.value }`)
      }
    }
    announce('Quantity Updated. New cart total: $' + getCartTotal(cart))
  }
}

export const cartUpdate = async (
  cart,
  skusNotAvailableIn,
  apiCalled,
  discount,
  setCartState,
  setApiCalled,
  updateOrder = false
) => {
  const region = getRegionZone().region
  if (cart && cart.cartItems && region) {
    checkProductRegionAvailability().then(skusNotAvailable => {
      if (skusNotAvailableIn.length === 0 || skusNotAvailable.length === 0) {
        setCartState({
          skusNotAvailable: skusNotAvailable,
        })
      }
    })
    let order = getOrder()
    const location = getCurrentLocation()
    const lineItems = getLineItems()
    let verifyQuantities
    if (order && order.lineItems) {
      verifyQuantities = compareLineItems(lineItems, order.lineItems)
    }
    const orderStorage = getFromBrowserStorage('session', 'order')
    if (!orderStorage) {
      order && store.dispatch(setOrder(order))
    }
    if (
      location &&
      lineItems &&
      lineItems.length > 0 &&
      (!order.orderId || updateOrder || verifyQuantities === false)
    ) {
      if (!apiCalled) {
        setApiCalled(true)
        if (!order.orderId) {
          order = await createOrder({
            lineItems: lineItems,
            region: location.region,
            zone: parseInt(location.price_zone),
            distribution_index: parseInt(location.distribution_index),
          })
          store.dispatch(setCheckoutStep('shipping'))
        } else {
          order = await updateLineItems({
            orderId: order.orderId,
            lineItems: lineItems,
            region: location.region,
            zone: parseInt(location.price_zone),
            distribution_index: parseInt(location.distribution_index),
          })
          order.total = order.total - order.tax - order.totalDeliveryCharge
          order.totalDeliveryCharge = 0
          order.tax = 0
          store.dispatch(setCheckoutStep('shipping'))
        }
        setApiCalled(false)
      }
      order && store.dispatch(setOrder(order))
    }
    setDiscount(order, discount, setCartState)
  }
}

export const setDiscount = (order, discountIn, setCartState) => {
  let discount = 0
  if (order && order.promotions && order.promotions.totalSavings && order.promotions.totalSavings > 0) {
    discount = parseFloat(order.promotions.totalSavings)
  }
  if (discount !== discountIn) {
    setCartState({
      showPayPal: true,
      discount: discount,
    })
  } else {
    setCartState({
      showPayPal: true,
    })
  }
}

export const removeUnavailableItems = (setCartState, skusNotAvailableIn) => {
  let cart = getCart()
  if (cart && cart.cartItems) {
    for (let i = cart.cartItems.length - 1; i >= 0; i--) {
      let skusNotAvailable = []
      skusNotAvailableIn.map(avail => skusNotAvailable.push(avail.sku))
      if (skusNotAvailable.includes(cart.cartItems[i].sku)) {
        cart.cartItems.splice(i, 1)
      }
    }
    if (cart.cartItems && cart.cartItems.length === 0) {
      cart = { cartItems: [] }
    }
    for (let key in sessionStorage) {
      if (key !== 'shippingInfo' && key !== 'cart') sessionStorage.removeItem(key)
    }
    store.dispatch(setCart(cart))
    setCartState({
      skusNotAvailable: [],
    })
  }
}

export const loadCartFromOrder = async orderId => {
  fetchOrder({ orderId: orderId }).then(order => {
    let cartItems = []
    if (order && order.lineItems && order.lineItems.length > 0) {
      order.lineItems.map(item => {
        cartItems.push(
          fetchProductOrGC(item.sku).then(product => {
            return {
              sku: product.sku,
              quantity: item.quantity,
              product: cartProduct(product),
              price: productPrice(product),
              activeAddons: null,
            }
          })
        )
      })
      Promise.all(cartItems).then(cartItems => {
        store.dispatch(setCart({ cartItems: cartItems }))
      })
    }
  })
}

export const getPromoTargetSkus = () => {
  const order = getOrder()
  if (order && order.promotions && order.promotions.availableTargetSkus) {
    return order.promotions.availableTargetSkus
  }
}
