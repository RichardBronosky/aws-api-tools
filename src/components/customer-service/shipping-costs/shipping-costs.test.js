import React from 'react'
import renderer from 'react-test-renderer'
import ShippingCosts from './shipping-costs'
import * as shippingCostsHelper from '../../../lib/helpers/shipping-costs'

describe('ShippingCosts', () => {
  const testProps = {
    zipCode: '33584',
    errorMessage: null,
    setState: jest.fn(),
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<ShippingCosts { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with error message', () => {
    const tree = renderer.create(<ShippingCosts { ...{ ...testProps, errorMessage: 'test error' } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls handleSubmit when simulating form onSubmit', () => {
    const tree = renderer.create(<ShippingCosts { ...testProps } />)
    shippingCostsHelper.handleSubmit = jest.fn()
    tree.root.findByProps({ className: 'input-wrapper' }).props.onSubmit({ preventDefault: jest.fn() })
    expect(shippingCostsHelper.handleSubmit).toHaveBeenCalled()
  })

  it('calls setState when simulating input onChange', () => {
    const tree = renderer.create(<ShippingCosts { ...testProps } />)
    tree.root.findByProps({ className: 'zip-input' }).props.onChange({ target: { value: 'test' } })
    expect(testProps.setState).toHaveBeenCalled()
  })
})
