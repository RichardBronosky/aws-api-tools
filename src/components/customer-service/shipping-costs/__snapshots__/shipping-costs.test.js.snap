// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingCosts renders correctly with default props 1`] = `
<div
  className="shipping-costs"
>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      ESTIMATED SHIPPING COSTS
    </h1>
    <p
      className="category-text"
    >
      Enter the zip code where the furniture is being DELIVERED to see the delivery costs to that area
    </p>
    <form
      className="input-wrapper"
      onSubmit={[Function]}
    >
      <input
        className="zip-input"
        onChange={[Function]}
        placeholder="Enter ZIP code"
        value="33584"
      />
      <button
        className="submit-btn"
        type="submit"
      >
        Go
      </button>
    </form>
  </div>
  <div
    className="category"
  >
    <img
      alt="Delivery Trucks"
      role="presentation"
      src="//images.ctfassets.net/n86sd31cg9pg/2d4uJ6HKg8gIkSGoEYqIyw/aeadf3096828b82337e688714eb4b480/delivery-map2-bg.png"
    />
  </div>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      WHERE WE DELIVER
    </h1>
    <p
      className="category-text"
    >
      Rooms To Go will deliver and set up your furniture purchase to the following states: Alabama, Florida, Georgia, South Carolina, North Carolina, Tennessee, Mississippi, Texas, Virginia and Kentucky. In addition, we deliver to many areas in Oklahoma, Massachusetts, New York, New Jersey, Pennsylvania, Maryland, Delaware, Connecticut, Louisiana, Rhode Island and Washington, D.C.
    </p>
  </div>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      FREE UPS DELIVERY
    </h1>
    <p
      className="category-text"
    >
      While furniture delivery is not available to every state, many home accessories on roomstogo.com are shipped via UPS Standard Ground shipping. This merchandise is clearly identified and will be shipped anywhere in the Contiguous U.S. often at no shipping cost to the consumer.
    </p>
  </div>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      QUESTIONS AFTER DELIVERY
    </h1>
    <p
      className="category-text"
    >
      Call Customer Care Center at 1-800-766-6786 with any concerns.
    </p>
    <p
      className="category-text"
    >
      Hours:
    </p>
    <p
      className="category-text"
    >
      Monday - Friday 8:00am - 8:00pm Eastern
    </p>
    <p
      className="category-text"
    >
      Saturday - 8:00am - 6:00pm Eastern
    </p>
  </div>
</div>
`;

exports[`ShippingCosts renders correctly with error message 1`] = `
<div
  className="shipping-costs"
>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      ESTIMATED SHIPPING COSTS
    </h1>
    <p
      className="category-text"
    >
      Enter the zip code where the furniture is being DELIVERED to see the delivery costs to that area
    </p>
    <div
      className="error-message"
    >
      test error
    </div>
    <form
      className="input-wrapper"
      onSubmit={[Function]}
    >
      <input
        className="zip-input"
        onChange={[Function]}
        placeholder="Enter ZIP code"
        value="33584"
      />
      <button
        className="submit-btn"
        type="submit"
      >
        Go
      </button>
    </form>
  </div>
  <div
    className="category"
  >
    <img
      alt="Delivery Trucks"
      role="presentation"
      src="//images.ctfassets.net/n86sd31cg9pg/2d4uJ6HKg8gIkSGoEYqIyw/aeadf3096828b82337e688714eb4b480/delivery-map2-bg.png"
    />
  </div>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      WHERE WE DELIVER
    </h1>
    <p
      className="category-text"
    >
      Rooms To Go will deliver and set up your furniture purchase to the following states: Alabama, Florida, Georgia, South Carolina, North Carolina, Tennessee, Mississippi, Texas, Virginia and Kentucky. In addition, we deliver to many areas in Oklahoma, Massachusetts, New York, New Jersey, Pennsylvania, Maryland, Delaware, Connecticut, Louisiana, Rhode Island and Washington, D.C.
    </p>
  </div>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      FREE UPS DELIVERY
    </h1>
    <p
      className="category-text"
    >
      While furniture delivery is not available to every state, many home accessories on roomstogo.com are shipped via UPS Standard Ground shipping. This merchandise is clearly identified and will be shipped anywhere in the Contiguous U.S. often at no shipping cost to the consumer.
    </p>
  </div>
  <div
    className="category"
  >
    <h1
      className="category-header"
    >
      QUESTIONS AFTER DELIVERY
    </h1>
    <p
      className="category-text"
    >
      Call Customer Care Center at 1-800-766-6786 with any concerns.
    </p>
    <p
      className="category-text"
    >
      Hours:
    </p>
    <p
      className="category-text"
    >
      Monday - Friday 8:00am - 8:00pm Eastern
    </p>
    <p
      className="category-text"
    >
      Saturday - 8:00am - 6:00pm Eastern
    </p>
  </div>
</div>
`;
